src/nvme/mi.c

```diff
@@ -414,10 +414,21 @@ static int nvme_mi_verify_resp_mic(struct nvme_mi_resp *resp)
 	return resp->mic != ~crc;
 }
 
+__attribute__((weak)) void *nvme_mi_submit_entry(__u8 type, const void *hdr, size_t hdr_len,
```

Comment:
Why the `void *` here? it's always at least a `struct nvme_mi_msg_hdr`, right?

(hook implementations can then cast according to type, if necessary...)

Reply:
Yes so fixed the `void *` to `struct nvme_mi_msg_hdr *`. Thank you.

Reply:
> Yes so fixed the `void *` to `struct nvme_mi_msg_hdr *`. Thank you.

but now we've lost the `const`; is that intentional?

Reply:
No missed to remain the `const` so fixed the patch again.